sanic/mixins/routes.py

```diff
@@ -806,23 +806,21 @@ async def _static_request_handler(
         __file_uri__=None,
     ):
         # Merge served directory and requested file if provided
-        # Strip all / that in the beginning of the URL to help prevent python
-        # from herping a derp and treating the uri as an absolute path
         root_path = file_path = unquote(file_or_directory)
 
         if __file_uri__:
+            # Strip all / that in the beginning of the URL to help prevent python
+            # from herping a derp and treating the uri as an absolute path
             unquoted_file_uri = unquote(__file_uri__).lstrip("/")
 
-            file_uri_pp = PurePath(unquoted_file_uri)
-            if any(part == ".." for part in file_uri_pp.parts):
+            segments = unquoted_file_uri.split("/")
+            if any(segment == ".." for segment in segments):
                 raise BadRequest("Invalid URL")
```

Comment:
I think we need to reject anything containing `\` as Invalid URL too. If left unchecked, Windows would treat it as path separator.

Reply:
Only needed on win32 because Unix does support `\` character in the directory name.

Reply:
You could test for the OS but it would probably be faster (both in development and execution time) to just reject `if '\\' in unquoted_file_url` on all operating systems. It's not like anyone actually names their files with backslashes (or carriage returns, vertical tabs, alarm bells and various other strange characters that the UNIX standards permit).

Reply:
We can do `any(os.sep in segment for segment in segments)`. 

`os.sep` will be `\\` on win32 and be `/` on Unix-like OSs.